devtools/maven/src/main/java/io/quarkus/maven/BuildFileMojoBase.java

```diff
@@ -0,0 +1,133 @@
+package io.quarkus.maven;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.maven.model.Dependency;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.repository.RemoteRepository;
+
+import io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver;
+import io.quarkus.cli.commands.file.BuildFile;
+import io.quarkus.cli.commands.file.GradleBuildFile;
+import io.quarkus.cli.commands.file.MavenBuildFile;
+import io.quarkus.cli.commands.writer.FileProjectWriter;
+import io.quarkus.platform.tools.config.QuarkusPlatformConfig;
+
+public abstract class BuildFileMojoBase extends AbstractMojo {
+
+    @Parameter(defaultValue = "${project}")
+    protected MavenProject project;
+
+    @Component
+    protected RepositorySystem repoSystem;
+
+    @Parameter(defaultValue = "${repositorySystemSession}", readonly = true)
+    protected RepositorySystemSession repoSession;
+
+    @Parameter(defaultValue = "${project.remoteProjectRepositories}", readonly = true, required = true)
+    protected List<RemoteRepository> repos;
+
+    @Override
+    public void execute() throws MojoExecutionException {
+
+        validateParameters();
```

Comment:
@aloubyansky this one is too long too bug I think it's copied code right?

Reply:
Not really, I actually type it in :)